File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *



n = int(stdin.readline()[:-1])


ans = 0

for _ in range(n):
    words = stdin.readline()[:-1]
    
    if len(words) % 2 != 0:
        continue
    
    stk = []

    for s in words :
        
        if stk and stk[-1] == s:
            stk.pop()
        
        else:
            stk.append(s)
        
    
    if not stk:
        ans += 1

print(ans)

File: 백준/Silver/3986. 좋은 단어/좋은 단어.swift
import Foundation

let n = Int(readLine()!)!


var ans: Int = 0

for _ in 0..<n {
    let arr = Array(readLine()!)

    if arr.count == 1 {
        ans += 1
        continue
    }
    
    var stk:[Character] = [arr[0]]

    for i in 1..<arr.count {
        
        if !stk.isEmpty && stk.last! == arr[i] {
            let _ = stk.popLast()
        } else {
            stk.append(arr[i])
        }
        
    }
    
    if stk.isEmpty {
        ans += 1
    }
}

print(ans)
